### PR TITLE
Allow to gather environments list

### PR DESCRIPTION
**Pull request by Dranaxel**
As using the library I was really missing this feature

### CODE REVIEW

**Review by caidanw — APPROVED** [review]
LGTM thanks for the PR

**Review by caidanw — CHANGES_REQUESTED** [review]
One final thing we need to do before merging this PR is to update the version.

The version is located in `domino_environments/_version.py` and will propagate changes to `setup.py`

We need to update from `1.0.3` to `1.1.0`

File: domino_environments/_environments.py [hunk 1]
```diff
@@ -46,6 +46,9 @@ def __init__(self, host: str):
     def deployment_version(self):
         return self.host + "/version"
 
+    def get_environments_list(self):
```
Comment (caidanw) [inline domino_environments/_environments.py]: I would like to keep the method names consistent
```suggestion
    def environments_list(self):
```

File: domino_environments/_environments.py [hunk 2]
```diff
@@ -197,9 +200,14 @@ def deployment_version(self):
         return self.request_manager.get(url).json().get("version")
 
     def refresh_defaults(self):
+        self._environments_list = self.get_environments_list()
         self._default_environment = self.get_default_environment()
         self._default_details = self.get_revision_details(self._default_environment)
 
+    def get_environments_list(self):
+        url = self._routes.get_environments_list()
```
Comment (caidanw) [inline domino_environments/_environments.py]: Updating the usage from prior change.
```suggestion
        url = self._routes.environments_list()
```

**Review by caidanw — CHANGES_REQUESTED** [review]
Just realized there are some semantic changes that I'd like to see.

**Comment by Dranaxel** [comment]
Hi @mildmelon thanks for your feedback, I totally understand the remarks. I will take some time to fix it this WE I hope

**Review by caidanw — APPROVED** [review]
LGTM, thanks I'll merge now

**Pull request merged**